UnityProject/Assets/Scripts/Systems/Fluids/MixAndVolume.cs

```diff
@@ -130,12 +132,12 @@ public void Add(MixAndVolume mixAndVolume, bool changeVolume = true)
 
 			var returnMix = mix.Take(mix.Total * percentage);
 
-			var newOne = new float[gasMix.Gases.Length];
-			var removeNewOne = new float[gasMix.Gases.Length];
-			for (int i = 0; i < gasMix.Gases.Length; i++)
+			var newOne = new GasData();
+			var removeNewOne = new GasData();
+			foreach (var gasData in gasMix.GasesArray)
 			{
-				removeNewOne[i] = gasMix.Gases[i] * percentage;
-				newOne[i] = gasMix.Gases[i] * (1 - percentage);
+				removeNewOne.SetMoles(gasData.GasType, gasMix.GasData.GetGasMoles(gasData.GasType) * percentage);
+				newOne.SetMoles(gasData.GasType, gasMix.GasData.GetGasMoles(gasData.GasType) * (1 - percentage));
```

Comment:
TODO stick `gasMix.GasData.GetGasMoles(gasData.GasType)` in a var so it doesnt need to be gotten twice